### PR TITLE
[Fluid] Gear scheme clean up, rotation and rename

### LINKED ISSUE #6206: [FluidDynamics] gear_scheme.h is missing the SLIP condition implementation
@KratosMultiphysics/fluid-dynamics 

If the `bdf2` time scheme is used, no SLIP condition is applied since it is not implemented in the `GearScheme`.

### PR DESCRIPTION
**Pull request by rubenzorrilla**
This PR cleans up a bit (mainly removing the deprecated `OpenMPUtils`) the `GearScheme` (BDF2) in the `FluidDynamicsApplication`.

It also includes the rotation tools to enforce the slip conditions in a multi-freedom constraint fashion as we do in the rest of CFD schemes.

Finally, it also renames the class and the file to a more descriptive name (`GearScheme` to `BDF2TurbulentScheme`). 

closes #6206

### CODE REVIEW

**Comment by philbucher** [comment]
I would also like to review this, can you please wait for my review?

File: applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h [hunk 1]
```diff
@@ -62,14 +60,14 @@ namespace Kratos
 /**
  */
 template<class TSparseSpace,class TDenseSpace>
-class GearScheme : public Scheme<TSparseSpace, TDenseSpace>
+class BDF2TurbulentScheme : public Scheme<TSparseSpace, TDenseSpace>
```
Comment (philbucher) [inline applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h:63]: is it necessary to have `Turbulent` in the name?
Seems a bit superfluous to me

File: applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h [hunk 2]
```diff
@@ -81,17 +79,15 @@ class GearScheme : public Scheme<TSparseSpace, TDenseSpace>
...
     ///@}
     ///@name Life Cycle
     ///@{
 
     /// Default constructor.
-    GearScheme()
-        :
-        Scheme<TSparseSpace, TDenseSpace>(),
-        mrPeriodicIdVar(Kratos::Variable<int>::StaticObject())
+    BDF2TurbulentScheme(const unsigned int DomainSize)
```
Comment (philbucher) [inline applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h]: why passing DOMAIN_SIZE from outside and not using the ProcessInfo like we usually do?

File: applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h [hunk 3]
```diff
@@ -752,19 +724,19 @@ class GearScheme : public Scheme<TSparseSpace, TDenseSpace>
...
                 iNode->FastGetSolutionStepValue(NODAL_AREA) = 1.0;
             }
             const double Area = iNode->FastGetSolutionStepValue(NODAL_AREA);
             iNode->FastGetSolutionStepValue(ADVPROJ) /= Area;
             iNode->FastGetSolutionStepValue(DIVPROJ) /= Area;
         }
 
-        KRATOS_INFO_IF("GearScheme", rModelPart.GetCommunicator().MyPID() == 0)
-            << "Computing OSS projections" << std::endl;
+        KRATOS_INFO("BDF2TurbulentScheme") << "Computing OSS projections" << std::endl;
```
Comment (philbucher) [inline applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h:747]: This is very verbose, can you protect it with an `echo_level`?

File: applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h (hunk 1 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h:63]: I think it is. Not all the schemes allow the use of a turbulent model.

File: applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h (hunk 2 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h]: Makes sense. I'll try to remove it.

File: applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h (hunk 3 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h:747]: OK.

File: applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h (hunk 2 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h]: Now I see. The point is that at construction time the `scheme` doesn't know about the model part. Hence we need to provide the domain size as an argument to set up the corresponding rotation tool.

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h]: Maybe we can set up the rotation tool in the initialize to remove this. :thinking:

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h]: Done.

File: applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h (hunk 3 above)
Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/custom_strategies/schemes/bdf2_turbulent_scheme.h:747]: I checked the base class and there is no echo level in the  `Scheme`. Probably we should add it in there before.

**Comment by rubenzorrilla** [comment]
@philbucher any other comment / suggestion?

File: applications/pfem_2_application/python_scripts/pfem_2_navier_stokes_monolithic_solver.py [hunk 4]
```diff
@@ -60,7 +60,7 @@ def Initialize(self):
                                         self.settings["move_mesh_strategy"].GetInt(),
                                         self.computing_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE])
             elif self.settings["time_scheme"].GetString() == "bdf2":
-                self.time_scheme = KratosCFD.GearScheme()
+                self.time_scheme = KratosCFD.BDF2TurbulentScheme(self.computing_model_part.ProcessInfo[KratosMultiphysics.DOMAIN_SIZE])
```
Comment (philbucher) [inline applications/pfem_2_application/python_scripts/pfem_2_navier_stokes_monolithic_solver.py]:
```suggestion
                self.time_scheme = KratosCFD.BDF2TurbulentScheme()
```

**Review by philbucher — DISMISSED** [review]
looks ok to me

maybe you want to wait for the approval of @KratosMultiphysics/fluid-dynamics

**Review by philbucher — APPROVED** [review]

**Comment by rubenzorrilla** [comment]
> looks ok to me
> 
> maybe you want to wait for the approval of @KratosMultiphysics/fluid-dynamics

Thanks. I'll wait a few days in case there are complaints.

**Comment by rubenzorrilla** [comment]
After one month, I think it's time to merge this.

**Pull request merged**